### ISSUE TITLE
LoadError: cannot load such file -- active_job_status/redis

### ISSUE THREAD

**Issue by maia** [cdale77/active_job_status#9]
README.md mentions to use:
````ruby
require "active_job_status/redis"
````
…but adding this to the class I'm creating jobs in will lead to the error:
````
LoadError: cannot load such file -- active_job_status/redis
	from …/active_support/dependencies.rb:274:in `require'
	from …/active_support/dependencies.rb:274:in `block in require'
	from …/active_support/dependencies.rb:238:in `block in load_dependency'
	from …/active_support/dependencies.rb:647:in `new_constants_in'
	from …/active_support/dependencies.rb:238:in `load_dependency'
	from …/active_support/dependencies.rb:274:in `require'
````
I'm using Rails 4.2.5, Ruby 2.2.2, Sidekiq 4.0.1, Redis 3.2.2 and ActiveJobStatus 0.0.5.

I believe the problem is that the documentation references a newer build than what's released as a gem (version 0.0.5 is from March 2015).

**Comment by maia** [comment]
When using the current master branch, the error changes to:
````
NameError: uninitialized constant ActiveSupport::Cache::RedisStore
	from …/bundler/gems/active_job_status-5dc60dd25571/lib/active_job_status/redis.rb:17:in `<top (required)>'
````
Btw, the common way to use redis via sidekiq is:
````ruby
Sidekiq.redis do |conn|
  conn.get(…)
end
````
…which will also make sure it's not opening additional connections, but using one of the connection pool.

**Comment by cdale77** [comment]
Hi @maia thanks for letting me know! Sorry you're having troubles. We tried to reduce the explicit dependency on Redis, and we'd like to make this useable for any ActiveJob backend, not just Sidekiq. 

I'll take a look at this soon. For now would it work for you to just use ActiveSupport::Cache::MemoryStore instead of Redis?

**Comment by maia** [comment]
I can use ActiveSupport::Cache::MemoryStore in development, but I'm reluctant to use it in production as it's not persistent and could cause problems when e.g. deploying in the wrong moment.

But something is confusing me: you mentioned you want to make it usable for any ActiveJob backend (available adapters are listed [here](http://api.rubyonrails.org/classes/ActiveJob/QueueAdapters.html)), why then reference redis at all? Which adapter does the current setup work with? ActiveSupport::Cache::RedisStore seems to require its own gem?

**Comment by cdale77** [comment]
My original implementation required Redis, as I generally use Sidekiq. Then another contributor modified the library to use ActiveSupport::Cache, so that we could remove Redis as a dependency. The idea was to allow users who already use Redis to use it as there are some nice optimizations available, but it should be optional.

ActiveSupport::Cache::RedisStore, I believe, is part of ActiveSupport, so nothing extra should be needed. I'm not sure why you are having problems, I will take a look as soon as I get a chance. 

One idea -- as you can see the base implementation is pretty trivial, using ActiveJob's callbacks. So you might be able go make something that works a bit better for your use case just by checking out what I did and modifying it a bit.

**Comment by maia** [comment]
Thanks. You're right, I've now rolled a small custom module I can include in my worker class that will save a redis set of job_ids for a given batch key, remove the set item on after_perform and also check if the set is empty, and if yes call another worker. See this [gist](https://gist.github.com/maia/a00974bf70875fef6b13).

**Comment by cdale77** [comment]
@maia Nice gist, thanks for sharing!

I looked into your problem today and could replicate it. The issue was that I didn't have ActiveSupport::Cache set up to use Redis. Once that was set up everything worked fine. 

In the process of this I set up a RailTie to automatically configure the Redis monkeypatch, so users won't have to explicitly require it anymore. The Readme is a little vague about using Redis, so I improved it as well. 

This work is here: https://github.com/cdale77/active_job_status/pull/11

I'm planning on releasing a new version very shortly, and I'll include these improvements. Thanks for filing the ticket that helped me track them down!
